### ISSUE TITLE
Route "Not Found" if using `route_prefix`

### ISSUE THREAD

**Issue by akadusei** [luckyframework/lucky#1281]
I get a `404` response using `ApiClient` to visit an action that uses `route_prefix`.

```crystal
abstract class ApiAction < Lucky::Action
  # ...
  route_prefix "/api/v0" # <= Works OK if this line is commented out
  #...
end

class Api::Posts::Index < ApiAction
  # ...
end

response = ApiClient.exec(Api::Posts::Index)
response.status_code # <= 404
```

It seems `Lucky::Routable.route` does not factor in the route prefix?

**Comment by jwoertink** [comment]
wow... Good catch.. That is indeed the case. We add the route [here](https://github.com/luckyframework/lucky/blob/master/src/lucky/routable.cr#L192), but then build all the helpers and everything off [this part](https://github.com/luckyframework/lucky/blob/master/src/lucky/routable.cr#L194) which omits the prefix. So what we would need is some sort of deal like

`{% path = ROUTE_SETTINGS[:prefix] + path %}`, and I think that's it....

**Comment by paulcsmith** [comment]
Hmm I wonder why the tests worked. Or maybe we just missed those tests? Either way, thanks for the bug report @akadusei

**Comment by jwoertink** [comment]
Yeah, we never covered that in the tests.

**Comment by paulcsmith** [comment]
Whoops!

**Comment by VinyLimaZ** [comment]
Hey guys, I stumbled in that same bug today, so IDK if this bug was really fixed...

### What I'm trying to do?
- I'm trying to test an Api endpoint.

### The code:
The action:
```
class Api::Crawler::WikipediaCollege::Conferences::Update < ApiCrawlerAction
  post "/wikipedia_college/conferences" do
```
The abstract class of the action:
```
abstract class ApiCrawlerAction < Lucky::Action
  route_prefix "/data"
```

Inside the test:
```
puts Api::Crawler::WikipediaCollege::Conferences::Update.route.path
=> /wikipedia_college/conferences
```

The expected return of the `Api::Crawler::WikipediaCollege::Conferences::Update.route.path`:
`/data/wikipedia_college/conferences`

**Comment by jwoertink** [comment]
My guess is it's related to the `route_prefix` being in the abstract class. I would have guessed this should work, but I don't see any specs to test that theory.

**Comment by matthewmcgarvey** [comment]
@VinyLimaZ Any chance you were still on Lucky 0.24? This change wasn't released until 0.25 (Dec 18, 2020)... and I'm kind of stumped if there's still a bug in the codebase 🙈

**Comment by matthewmcgarvey** [comment]
Closing for now
